### PR TITLE
feat: replace x-button with XDS x-button implementation

### PR DESCRIPTION
**Pull request by CachedaCodes**
EX-5908

## Motivation and context
Add `x-button` implementation from the XDS

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Change requires a documentation update

## What is the destination branch of this PR?

- [X] `Main`
- [ ] Other. Specify:

### CODE REVIEW

**Review by herrardo — CHANGES_REQUESTED** [review]
I am comparing this PR Preview with what we have deployed in test. I see some differences.

1. There is different margin between suggestions, i guess because history queries have a button inside
image

2. Have we change the font family? I see different behavior
image

3. The difference between the magnifying glass button on the search input is on purpose?

image

**Comment by CachedaCodes** [comment]
@herrardo 

> I am comparing this PR Preview with what we have deployed in test. I see some differences.
> 
> 1. There is different margin between suggestions, i guess because history queries have a button inside

Yes, a quick fix was made in the PR, and we'll deal with it when we implement the suggestion group

> 2. Have we change the font family? I see different behavior

The one here is Montserrat, which I think is correct, the one in x.test is default sans-serif for some reason. Anyway, it should be Montserrat.

> 3. The difference between the magnifying glass button on the search input is on purpose?

Fixed, thanks!

**Comment by herrardo** [comment]
> @herrardo
> 
> > I am comparing this PR Preview with what we have deployed in test. I see some differences.
> > 
> > 1. There is different margin between suggestions, i guess because history queries have a button inside
> 
> Yes, a quick fix was made in the PR, and we'll deal with it when we implement the suggestion group
> 
> > 2. Have we change the font family? I see different behavior
> 
> The one here is Montserrat, which I think is correct, the one in x.test is default sans-serif for some reason. Anyway, it should be Montserrat.
> 
> > 3. The difference between the magnifying glass button on the search input is on purpose?
> 
> Fixed, thanks!

I think the font-family is ok in test environment, we are using **Inter** font, but ask @cris-crash to be sure.

**Comment by CachedaCodes** [comment]
There are known issues with the XDS button implementation that I'm currently discussing with design. I will deal with them afterwards.

Now that the `x-button` has hover effects, the size created inside of the button by its padding is noticeable in the `x-button-ghost`. In the past we used the padding of the button to align it with other elements when needed (removing it, for example), doing that now creates this effect

image

Clear button in the searchbox is part of an input group that forces some styles (same bg color for example). The `x-button` class adds an on hover effect that changes colors, resulting in this effect when hovering the clear button.

image

Sometimes, the new hover effects and its interaction with the `x-button-ghost` produce weird results depending on the background the button is or the elements inside the icon.

image

image

The button size changing makes it so elements previously aligned now are not.

image

Some combinations of bg color and font are not possible with the DS or clash with it

image

image

File: src/components/search/results/mobile-partial-results.vue [hunk 1]
```diff
@@ -18,15 +18,15 @@
         </span>
         <PartialQueryButton
           :query="partialResult.query"
-          class="x-text x-font-weight--bold x-button--ghost x-margin--left-auto"
+          class="x-button-lead x-button-ghost x-margin--left-auto"
```
Comment (tiborux) [inline src/components/search/results/mobile-partial-results.vue:21]: Removing the x-text the font-size changes, I supposed that this is right

**Review by tiborux — CHANGES_REQUESTED** [review]
gj

**Comment by tiborux** [comment]
You missed the next-query:

image

**Review by mnavarroespinosa — CHANGES_REQUESTED** [review]
Good job!

**Comment by CachedaCodes** [comment]
> You missed the next-query:
> 
> image

This one is technically a suggestion, which will be its own component, so it shouldn't be changed.

**Review by tiborux — APPROVED** [review]
gj

**Review by mnavarroespinosa — APPROVED** [review]

**Review by tiborux — APPROVED** [review]

**Pull request merged**